src/Http.php

```diff
@@ -34,8 +33,8 @@ public function withRateLimiting(): self
     }
 
     public function get(
-        Endpoint $endpoint,
-        string|array|Body $body,
+        EndpointInterface $endpoint,
```

Comment:
```suggestion
        EndpointInterface|string $endpoint,
```
Some endpoints such as [`/messages`](https://github.com/dc-Ragnarok/Bifrost/blob/aa1fc89889ffd75bff566b8cd73839ff1e4fa6d7/src/Const/DiscordEndpoints.php#L53) do not need to be converted into an instance of Endpoint.

Reply:
`/messages` is a suffix for channels etc, which do have an ID requirement?

Regardless,
Having a union type would increase complexity as it has to be checked whether it's a string or not. This would also have to be supported in the middleware driver etc.

I dont think its worth it to introduce said complexity when an endpoint without params can also be used in `Endpoint`.

Admittedly theres no test case for such endpoint, so I'd suggest we add a this test case and dont add the union type.

Is this something youre ok with?

Reply:
I did only now just realize that we can also convert the string to an `Endpoint` in the `Request` function so it's all an `Endpoint` from there on. Lmk what you prefer

Reply:
That's fine, I know [`/gateway/bot`](https://github.com/dc-Ragnarok/Bifrost/blob/feature/requests/src/Const/DiscordEndpoints.php#L22) doesn't require params but since most of the others ones do I guess it really isn't worth adding a union type. I'm presuming the user can still use `Endpoint::bind` with `/gateway/bot`

Reply:
Yeah thats a supported usecase, just no explicit test case for it yet, which I'll add